### DISCUSSION TITLE
Package Manager support private repositories?

### DISCUSSION THREAD

**Discussion by CThuleHansen** [OpenModelica/OpenModelica#11204] [Q&A] [answered]
Hi,
is it possible to enable the functionality of the package manager for private repositories? 
If I add a private repository to `index.json`, then it is removed when I call `updatePackageIndex`.

From what I an see in https://github.com/OpenModelica/OpenModelica/blob/bf24d82543a633f42f010a1fc39a3e76cd95d3c1/OMCompiler/Compiler/Script/PackageManagement.mo it does not seem to be the case.

If it is not possible. does anyone have some good practices concerning this?

**Comment by sjoelund** [comment, marked as answer]
It's not possible at the moment. What would be nice to have is the ability to configure multiple URLs for indexes, but OpenModelica (OMC) itself does not have configuration files (only OMEdit does).

**Reply by CThuleHansen** [reply]
Thank you. Can you elaborate on:
> What would be nice to have is the ability to configure multiple URLs for indexes, but OpenModelica (OMC) itself does not have configuration files (only OMEdit does).

?
I cannot follow your trail of thought.

**Reply by sjoelund** [reply]
So, to change the code internally in OpenModelica to:
1. Handle reading multiple index.json files when performing operations (so you can have a file for manual changes)
2. Downloading these from multiple URLs (so you can share the index with colleagues)

But since OMC does not have any settings anywhere (except in environment variables), this is a bit more work than just adding the above functionality.

**Reply by CThuleHansen** [reply]
Okay - Thank you for elaborating.
How about storing a file next to the `index.json` file, since this file and its location is already known to the tooling?
